synth/src/cli/telemetry.rs

```diff
@@ -13,6 +19,60 @@ use super::{Args, TelemetryCommand};
 const API_KEY: &str = "L-AQtrFVtZGL_PjK2FbFLBR3oXNtfv8OrCD8ObyeBQo";
 const EVENT_NAME: &str = "synth-command";
 
+lazy_static! {
+    static ref TELEMETRY_CLIENT: TelemetryClient = TelemetryClient::new();
+}
+
+fn report_panic(synth_command: &str, telemetry_client: &TelemetryClient) {
+    // This exists only to return early in case of a panic inside this panic hook
+    let report_panic_impl = || -> Result<()> {
+        let (stdin, stdout) = (io::stdin(), io::stdout());
+        let (mut stdin, mut stdout) = (stdin.lock(), stdout.lock());
+        let mut username = None;
+        let mut email = None;
+
+        println!("{}", "Synth encountered a fatal error. Because telemetry is enabled, an anonymous \
```

Comment:
Isn't it better if those are going to stderr instead?

Reply:
That makes sense. But only this first message should go to stderr, right? Otherwise, if the user, e.g., redirects stderr, they wouldn't see the other messages that wait for a response.

Reply:
That's a good point! Actually we should check `synth` is being run interactively before this otherwise this will deadlock.

This can be done with something like [`console::user_attended_stderr`](https://docs.rs/console/0.14.1/console/fn.user_attended_stderr.html) for an easy OOB solution.

I would still say it should all go to `stderr`, as this is, I think, where most people would expect diagnostic output to be sent.

Reply:
Right, I haven't thought of that. I will follow your suggestions to fix the issue.